synapse/handlers/message.py

```diff
@@ -1235,7 +1262,12 @@ async def _send_dummy_event_for_room(self, room_id: str) -> bool:
                 # Since this is a dummy-event it is OK if it is sent by a
                 # shadow-banned user.
                 await self.send_nonmember_event(
-                    requester, event, context, ratelimit=False, ignore_shadow_ban=True,
+                    requester,
+                    event,
+                    context,
+                    ratelimit=False,
+                    ignore_shadow_ban=True,
+                    ignore_third_party_event_rules=True,
```

Comment:
why are we doing this?

Reply:
Ah, yes it's not clear from this PR at all, sorry.

The overall intention here is to allow a `ThirdPartyEventRules` module to be able to send events into a room. `ThirdPartyEventRules` can also listen to events coming into a room, and make sure they're allowed (via `check_event_allowed`). If a `ThirdPartyEventRules` module sent an event, presumably it doesn't need to then check that event is allowed to be sent.

In addition to fixing the circular dependency, this also bypasses that unnecessary situation.

Reply:
riight, but this is a dummy event, which we know was *not* send by a `ThirdPartyEventRules` module.

Reply:
Ahh, this is the dummy event sending bit. I thought you were talking generally about `ignore_third_party_event_rules`.

I originally did this as I didn't think we'd need to block this type of event by `ThirdPartyEventRules`, but I realise there could be a usecase where we get notified about events like this via our module.

So this probably doesn't belong here, no.

Reply:
We're stripping these options. See https://github.com/matrix-org/synapse/pull/8479#discussion_r501811890